test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace BDBStudios\BreakfastSerializerTests\Property\TypeHandler;
+
+
+use BDBStudios\BreakfastSerializer\Property\TypeHandler\IsDateTime;
+use BDBStudios\BreakfastSerializerTest\Fixtures\TypeHandlers\DateTimeType;
+
+class DateTimeHandler extends \PHPUnit_Framework_TestCase
+{
+    /** @var  DateTimeType */
+    protected $instance;
+
+    const ISO_REGEX = '/^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})((-|\+)\d{4})$/';
```

Comment:
I appreciate the concatenation,  though the long form is so perverted I may
have to use it
On 28 Jul 2015 8:45 PM, "François Pietka" <notifications@github.com> wrote:

> In test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php
> <https://github.com/brettminnie/breakfast-serializer/pull/34#discussion_r35691579>
> :
>
> > @@ -0,0 +1,51 @@
> > +<?php
> > +
> > +namespace BDBStudios\BreakfastSerializerTests\Property\TypeHandler;
> > +
> > +
> > +use BDBStudios\BreakfastSerializer\Property\TypeHandler\IsDateTime;
> > +use BDBStudios\BreakfastSerializerTest\Fixtures\TypeHandlers\DateTimeType;
> > +
> > +class DateTimeHandler extends \PHPUnit_Framework_TestCase
> > +{
> > +    /** @var  DateTimeType */
> > +    protected $instance;
> > +
> > +    const ISO_REGEX = '/^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})((-|\+)\d{4})$/';
>
> How about:
>
> /^(19[0-9]{2}|[2-9][0-9]{3})-((0(1|3|5|7|8)|10|12)-(0[1-9]|1[0-9]|2[0-9]|3[0-1])|(0(4|6|9)|11)-(0[1-9]|1[0-9]|2[0-9]|30)|(02)-(0[1-9]|1[0-9]|2[0-9]))T(0[0-9]|1[0-9]|2[0-3])(:[0-5][0-9]){2}[-+]((0[0-9]|1[0-3])([03]0|45)|14:00)$/
>
> Doesn't check leap year though.
>
> Kidding [image: :grin:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brettminnie/breakfast-serializer/pull/34/files#r35691579>
> .
>